src/main/java/org/broadinstitute/hellbender/engine/ReadsDataSource.java

```diff
@@ -288,8 +288,21 @@ public SAMFileHeader getHeader() {
 
         // Set up an iterator for each reader, bounded to overlap with the supplied intervals if there are any
         for ( Map.Entry<SamReader, CloseableIterator<SAMRecord>> readerEntry : readers.entrySet() ) {
-            readerEntry.setValue(traversalIsBounded ? new SamReaderQueryingIterator(readerEntry.getKey(), queryIntervals, queryUnmapped) :
-                                                      readerEntry.getKey().iterator());
+            if (traversalIsBounded) {
+                if (readers.size() > 1) {
+                    readerEntry.setValue(
```

Comment:
could you pull the `readerEntry.setValue` call out of the statement?

 i.e. have a tenary expression that returns either `queryIntervals` or `getIntervalsOverlappingReader(queryIntervals)` followed by the call to `readerEntry.setValue(...)`

that would make it clearer what the difference between the two branches is at quick inspection